### DISCUSSION TITLE
Require PHP (minor) version

### DISCUSSION THREAD

**Discussion by mbrodala** [renovatebot/renovate#11954] [Request Help] [answered]
Currently Renovate evaluates the `php` version constraint in the `composer.json` and uses that to determine the newest possible version of its php image.

So given a version constraint like `"php": "^7.2"` Renovate will do updates with PHP 7.4.x. However, this completely skips the PHP version requirements of installed dependencies which will lead to many errors like this:

```
<package> <version> requires php ^7.0 < 7.3 -> your php version (7.4.24) does not satisfy that requirement.
```

So far we have only configured `"composerIgnorePlatformReqs": [ "ext-intl" ]` to get updates running.

Any idea how to solve this? Can Renovate be taught to prefer a specific PHP minor version?

**Comment by viceice** [comment]
Shouldn't `~` range do it?

**Comment by viceice** [comment, marked as answer]
You need `~7.2.0`

https://getcomposer.org/doc/articles/versions.md#tilde-version-range-

**Reply by mbrodala** [reply]
(Damn Github dropped my answer as I clicked "Mark as answer")

You are absolutely right, switching from `^7.2` to `~7.2.0` solved the issue, Renovate now uses its `docker.io/renovate/php:7.2.34` image.

And it indeed does not make sense to allow newer PHP minor versions in a project since it is not developed with and tested against anything but the minimum PHP minor version. We evidently had no issue so far since most projects use `^7.4` and there is no newer 7.x PHP minor version.
